src/react-apps/applications/runtime/src/utils/validation.ts

```diff
@@ -43,13 +44,44 @@ const validationFunctions: any = {
   pattern,
 };
 
+export function validateFormComponents(
+  formAttachments: any,
+  formLayout: any,
+  language: any,
+) {
+  const validations = {};
+  const fieldKey = 'simpleBinding';
+  formLayout.forEach((component) => {
+    validations[component.id] = {};
+    if (component.type === 'FileUpload') {
+      if (component.minNumberOfAttachments > 0 && Object.keys(formAttachments).length < 1 ||
```

Comment:
Should probably include a null check for formAttachments? Object.keys(null) will crash.

Reply:
Object.keys(formAttachments) returns an empty array if there are no attachments. 
Default is:
formAttachments: {
attachments: {}
}, therefore it won't break.

Reply:
Then you are dependent on the state being initialized as not null, I still feel like this method should work independent on what input parameters you pass inn. Passing null values should not break a validation function imo.